src/components/Header/Header.tsx

```diff
@@ -0,0 +1,114 @@
+import './Header.css';
+import SearchBar from './SearchBar/SearchBar';
+import * as FaIcons from 'react-icons/fa';
+import { FaRegUserCircle, FaShoppingCart } from 'react-icons/fa';
+import { ReactElement, useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { SideBar } from '../SideBar/SideBar';
+import { openLoginModal, getTotalItemsInCart, logOut, addNewToast, NewToastParams } from '../../slices';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useRouteMatch } from 'react-router-dom';
+import { RootState } from '../../rootReducer';
+import { CSSTransition } from 'react-transition-group';
+
+export interface HeaderProps {
+	userName: string | null;
+	isLoggedIn: boolean;
+}
+
+export const Header = ({ userName, isLoggedIn }: HeaderProps): ReactElement => {
+	const [isScrollTop, setIsScrollTop] = useState(true);
+	const [sidebarHeader, setSidebarHeader] = useState(false);
+	const itemsInCart = useSelector((state: RootState) => getTotalItemsInCart(state));
+	const dispatch = useDispatch();
+	const history = useHistory();
+	const match = useRouteMatch('/');
+
+	const theme = match?.isExact ? 'white' : 'dark';
```

Comment:
I didn't see the theme toggle feature in your demo? 😄 
btw, shouldn't it be `light` and `dark` ? 🤔